public/app/features/alerting/unified/components/admin/AlertmanagerConfigSelector.tsx

```diff
@@ -0,0 +1,101 @@
+import { css } from '@emotion/css';
+import React, { useMemo } from 'react';
+
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { Button, Select, useStyles2 } from '@grafana/ui';
+import { AlertManagerCortexConfig } from 'app/plugins/datasource/alertmanager/types';
+
+import { alertmanagerApi } from '../../api/alertmanagerApi';
+import { GRAFANA_RULES_SOURCE_NAME } from '../../utils/datasource';
+
+import { FormValues } from './AlertmanagerConfig';
+import { ConfigEditor } from './ConfigEditor';
+
+export interface ValidAmConfigOption {
+  label?: string;
+  value?: AlertManagerCortexConfig;
+}
+
+interface AlertmanagerConfigSelectorProps {
+  onChange: (selectedOption: ValidAmConfigOption) => void;
+  selectedAmConfig?: ValidAmConfigOption;
+  defaultValues: FormValues;
+  onSubmit: (values: FormValues, oldConfig?: AlertManagerCortexConfig) => void;
+  readOnly: boolean;
+  loading: boolean;
+}
+
+export default function AlertmanagerConfigSelector({
+  onChange,
+  selectedAmConfig,
+  defaultValues,
+  onSubmit,
+  readOnly,
+  loading,
+}: AlertmanagerConfigSelectorProps): JSX.Element {
+  const { useGetValidAlertManagersConfigQuery, useResetAlertManagerConfigToOldVersionMutation } = alertmanagerApi;
+
+  const styles = useStyles2(getStyles);
+
+  const { currentData: validAmConfigs, isLoading: isFetchingValidAmConfigs } = useGetValidAlertManagersConfigQuery();
+
+  const [resetAlertManagerConfigToOldVersion] = useResetAlertManagerConfigToOldVersionMutation();
+
+  const validAmConfigsOptions = useMemo(() => {
+    if (!validAmConfigs?.length) {
+      return [];
+    }
+
+    const configs: ValidAmConfigOption[] = validAmConfigs.map((config) => ({
+      label: config.last_applied ? `Config from ${new Date(config.last_applied).toLocaleString()}` : 'Previous config',
+      value: config,
+    }));
+    onChange(configs[0]);
```

Comment:
What is the reason for this? What if the `configs` array is empty?

Reply:
It would call onChange with `undefined` and this would set `selectedAmConfig` to `undefined`

Reply:
I would say that this is correct as we set `selectedAmConfig` to undefined in case we don't have `validAmConfigs`

Reply:
> It would call onChange with undefined

That's my concern, as according to `AlertmanagerConfigSelectorProps` the `onChange` doesn't accept undefined
`onChange: (selectedOption: ValidAmConfigOption) => void;`

Reply:
This code wouldn't execute if `configs` is empty because we don't show the dropdown in that case (see https://github.com/grafana/grafana/pull/65746/files#diff-e1b066e38166296feb3c05add1abf23ab0c47dd362b404c91be28f749c067d9aR73), and this is the onChange handler for when the dropdown option is changed.